numpy/core/shape_base.py

```diff
@@ -177,6 +177,71 @@ def atleast_3d(*arys):
         return res
 
 
+def atleast_nd(n, *arys):
+    """
+    View inputs as arrays with at least `n` dimensions.
+
+    Parameters
+    ----------
+    n : scalar
```

Comment:
I don't disagree with the possibility that this is a design mistake. The function should only take one array argument. If you need to pass in a tuple or list, use an iterator or list comprehension or whatever. In fact, let me think about it and perhaps I will break away from the existing convention for the atleast functions.

The need for the function comes from a use-case that I ran across. Perhaps it is not that universal, but I think it will be useful nevertheless.